src/base.ts

```diff
@@ -82,12 +82,12 @@ export default abstract class extends Command {
   async finally(error: Error | undefined): Promise<any> {
     await super.finally(error);
     this.metricsMetadata['success'] = error === undefined;
-    this.metricsMetadata['source'] = this.generateSHA1();
+    this.metricsMetadata['source'] = await this.generateSHA1(this.specFile?.getSource() || '');
     await this.recordActionFinished(this.id as string, this.metricsMetadata, this.specFile?.text());
   }
   
-  generateSHA1(): any {
-    const fileBuffer = readFileSync(this.specFile?.getSource() || '');
+  async generateSHA1(filePath: string): Promise<any> {
```

Comment:
I don't want to be intrusive but md5 is way faster than sha1 and for this I think it will be enough.

Additionally, are you sure you don't have access already to the content of the file? I'm more than sure you will have it in memory already, so no need to read from filesystem again (which is slow since I/O is required)